pkg/reconciler/v1alpha1/broker/broker.go

```diff
@@ -0,0 +1,442 @@
+			if err != nil {
+				return nil, err
+			}
+		}
+
+		return c, nil
+	}
+}
+
+func (r *reconciler) InjectClient(c client.Client) error {
+	r.client = c
+	return nil
+}
+
+// Reconcile compares the actual state with the desired, and attempts to
+// converge the two. It then updates the Status block of the Broker resource
+// with the current status of the resource.
+func (r *reconciler) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	ctx := context.TODO()
+	ctx = logging.WithLogger(ctx, r.logger.With(zap.Any("request", request)))
+
+	broker := &v1alpha1.Broker{}
+	err := r.client.Get(ctx, request.NamespacedName, broker)
+
+	if errors.IsNotFound(err) {
+		logging.FromContext(ctx).Info("Could not find Broker")
+		return reconcile.Result{}, nil
+	}
+
+	if err != nil {
+		logging.FromContext(ctx).Error("Could not Get Broker", zap.Error(err))
+		return reconcile.Result{}, err
+	}
+
+	// Reconcile this copy of the Broker and then write back any status updates regardless of
+	// whether the reconcile error out.
+	result, reconcileErr := r.reconcile(ctx, broker)
+	if reconcileErr != nil {
+		logging.FromContext(ctx).Error("Error reconciling Broker", zap.Error(reconcileErr))
+	} else if result.Requeue || result.RequeueAfter > 0 {
+		logging.FromContext(ctx).Debug("Broker reconcile requeuing")
+	} else {
+		logging.FromContext(ctx).Debug("Broker reconciled")
+		r.recorder.Event(broker, corev1.EventTypeNormal, brokerReconciled, "Broker reconciled")
+	}
+
+	if _, err = r.updateStatus(broker.DeepCopy()); err != nil {
+		logging.FromContext(ctx).Error("Failed to update Broker status", zap.Error(err))
+		r.recorder.Eventf(broker, corev1.EventTypeWarning, brokerUpdateStatusFailed, "Failed to update Broker's status: %v", err)
+		return reconcile.Result{}, err
+	}
+
+	// Requeue if the resource is not ready:
+	return result, reconcileErr
+}
+
+func (r *reconciler) reconcile(ctx context.Context, b *v1alpha1.Broker) (reconcile.Result, error) {
+	b.Status.InitializeConditions()
+
+	// 1. Channel is created for all events.
+	// 2. Filter Deployment.
+	// 3. Ingress Deployment.
+	// 4. K8s Services that point at the Deployments.
+
+	if b.DeletionTimestamp != nil {
+		// Everything is cleaned up by the garbage collector.
+		return reconcile.Result{}, nil
+	}
+
+	c, err := r.reconcileChannel(ctx, b)
+	if err != nil {
+		logging.FromContext(ctx).Error("Problem reconciling the channel", zap.Error(err))
+		b.Status.MarkChannelFailed(err)
+		return reconcile.Result{}, err
+	} else if c.Status.Address.Hostname == "" {
+		logging.FromContext(ctx).Info("Channel is not yet ready", zap.Any("c", c))
+		// Give the Channel some time to get its address. One second was chosen arbitrarily.
+		return reconcile.Result{RequeueAfter: time.Second}, nil
+	}
+	b.Status.MarkChannelReady()
+
+	_, err = r.reconcileFilterDeployment(ctx, b)
+	if err != nil {
+		logging.FromContext(ctx).Error("Problem reconciling filter Deployment", zap.Error(err))
```

Comment:
Wouldn't it be helpful to mark FilterReady=False?

Reply:
Done.